### PR TITLE
feat(driver adapters): [public API] Rename `flavour` field to `provider`, remove `close()`

### LINKED ISSUE #21865: [Driver Adapters]: rename `flavour` public field to `provider`
Currently, each Driver Adapter exposes a public `flavour` field indicating the underlying compatible `provider` in `schema.prisma`.
In an effort to unify naming conventions, we should rename [`flavour`](https://github.com/prisma/prisma/blob/8fdb2bcb5b2944d9f4b6dd1a9efa20c88f0b5ab4/packages/driver-adapter-utils/src/types.ts#L66) to `provider` in the [`Queryable`](https://github.com/prisma/prisma/blob/8fdb2bcb5b2944d9f4b6dd1a9efa20c88f0b5ab4/packages/driver-adapter-utils/src/types.ts#L65-L84) interface (and any class that extend such interface).

This issue related to other tasks related to `flavour`/`provider`-based validation, such as: https://github.com/prisma/prisma/issues/21864.

### PR DESCRIPTION
**Pull request by jkomyno**
This PR closes https://github.com/prisma/prisma/issues/21865 and closes https://github.com/prisma/team-orm/issues/574.

These changes affect the `DriverAdapter` interface from `@prisma/driver-adapter-utils`:
- `flavour: string` is now `provider: string`
- `close(): Promise<Result<void>>` is no longer required by the public API

### CODE REVIEW

File: packages/driver-adapter-utils/src/types.ts [hunk 1]
```diff
@@ -96,7 +96,7 @@ export interface DriverAdapter extends Queryable {
   /**
    * Closes the connection to the database, if any.
    */
-  close: () => Promise<Result<void>>
+  close?: () => Promise<Result<void>>
```
Comment (SevInf) [inline packages/driver-adapter-utils/src/types.ts:99]: I think we can remove it completely?

File: packages/client/package.json [hunk 2]
```diff
@@ -104,7 +104,7 @@
     "@prisma/internals": "workspace:*",
     "@prisma/migrate": "workspace:*",
     "@prisma/mini-proxy": "0.9.4",
-    "@prisma/query-engine-wasm": "0.0.16",
+    "@prisma/query-engine-wasm": "0.0.18",
```
Comment (jkomyno) [inline packages/client/package.json:107]: TODO: remove this

**Review by SevInf — APPROVED** [review]

**Review by aqrln — COMMENTED** [review]
Could you please also remove the `close` method from libSQL adapter? If it's not required by the engine, let's not add extra things to the public API surface, users can call `client.close()` directly

**Review by aqrln — APPROVED** [review]

**Comment by aqrln** [comment]
@shiyuhang0 5.7.0 tomorrow

**Pull request merged**